### PR TITLE
Implement slogdet

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented slogdet which calculates sign and log of determinant.

### CODE REVIEW

File: cupy/linalg/norms.py [hunk 1]
```diff
@@ -123,7 +131,87 @@ def norm(x, ord=None, axis=None, keepdims=False):
...
+    getrf(handle, m, m, a_copy.data.ptr, m, workspace.data.ptr,
+          ipiv.data.ptr, info.data.ptr)
+
+    if info[()] == 0:
+        diag = cupy.diag(a_copy)
+        # ipiv is 1-origin
+        non_zero = (cupy.count_nonzero(ipiv != cupy.arange(1, m + 1)) +
+                    cupy.count_nonzero(diag < 0))
+        if non_zero % 2 == 0:
+            sign = dtype.type(1.0)
```
Comment (okuta) [inline cupy/linalg/norms.py]: Please return `cupy.ndarray`.

File: cupy/linalg/norms.py [hunk 2]
```diff
@@ -123,7 +131,87 @@ def norm(x, ord=None, axis=None, keepdims=False):
...
+        else:
+            sign = dtype.type(-1.0)
```
Comment (okuta) [inline cupy/linalg/norms.py]: Please return `cupy.ndarray`.

File: cupy/linalg/norms.py [hunk 3]
```diff
@@ -123,7 +131,87 @@ def norm(x, ord=None, axis=None, keepdims=False):
...
+        logdet = cupy.log(abs(diag)).sum()
+    else:
+        sign = dtype.type(0.0)
```
Comment (okuta) [inline cupy/linalg/norms.py]: Please return `cupy.ndarray`.

File: cupy/linalg/norms.py [hunk 4]
```diff
@@ -123,7 +131,87 @@ def norm(x, ord=None, axis=None, keepdims=False):
...
+    workspace = cupy.empty(buffersize, dtype=dtype)
+    getrf(handle, m, m, a_copy.data.ptr, m, workspace.data.ptr,
+          ipiv.data.ptr, info.data.ptr)
+
+    if info[()] == 0:
+        diag = cupy.diag(a_copy)
+        # ipiv is 1-origin
+        non_zero = (cupy.count_nonzero(ipiv != cupy.arange(1, m + 1)) +
+                    cupy.count_nonzero(diag < 0))
+        if non_zero % 2 == 0:
```
Comment (okuta) [inline cupy/linalg/norms.py]: How about removing this branch for non-blocking operation.

File: cupy/linalg/norms.py (hunk 1 above)
Comment (unnonouno) [inline cupy/linalg/norms.py]: Do you mean zero-dimensional array?

**Comment by unnonouno** [comment]
Do i need to fix `count_nonzero` to return `ndarray` first?

**Comment by okuta** [comment]
LGTM!

**Pull request merged**